src/async_action.cpp

```diff
@@ -264,13 +264,24 @@ void AsyncHandlers::printf(const OpaqueValue &data)
             static_cast<uint64_t>(AsyncAction::printf);
   auto &fmt = std::get<0>(bpftrace.resources.printf_args[id]);
   auto &args = std::get<1>(bpftrace.resources.printf_args[id]);
+  auto severity = std::get<2>(bpftrace.resources.printf_args[id]);
+  auto &source_info = std::get<3>(bpftrace.resources.printf_args[id]);
   auto vals = prepare_args(
       bpftrace, c_definitions, args, data.slice(sizeof(uint64_t)));
   if (!vals) {
     LOG(BUG) << "Error processing printf arguments: " << vals.takeError();
   }
 
-  out.printf(fmt.format(*vals));
+  switch (severity) {
+    case PrintfSeverity::NONE: {
```

Comment:
Optional: I think that this severity also makes sense and something defined by the output (e.g. in `output.h`). You could have a single `printf` method that accepts that enum, rather than another. I think there's an argument either way however, so I don't have a hard preference. Just thinking out loud.

Reply:
Yeah I also went back and forth on that. They seemed different enough that I wanted a more explicit separate function call but YMMV.